src/main/scala/beam/agentsim/infrastructure/ChargingNetworkManager.scala

```diff
@@ -91,6 +86,17 @@ class ChargingNetworkManager(
         s"timeSpentToPlanEnergyDispatchTrigger: ${timeSpentToPlanEnergyDispatchTrigger} ms, nHandledPlanEnergyDispatchTrigger: ${nHandledPlanEnergyDispatchTrigger}, AVG: ${timeSpentToPlanEnergyDispatchTrigger.toDouble / nHandledPlanEnergyDispatchTrigger}"
       )
 
+    case inquiry: ParkingInquiry =>
+      log.debug(s"Received parking inquiry: $inquiry")
+      chargingNetworkMap(inquiry.vehicleManagerId).processParkingInquiry(inquiry) match {
+        case Some(parkingResponse) => {
+          sender() ! parkingResponse
+        }
+        case _ => {
+          Future(parkingNetworkManager ? inquiry).pipeTo(sender())
```

Comment:
I still think this is a bug. The ask already returns a future. So, this is not sending the response to the sender, but instead sending a future of the response...which will probably be thrown away as not matching...

Reply:
After this is addressed, the rest of my comments could be handled in a separate PR (or not at all), and will mark for approval thereafter. But I think this one to be a bug...

Reply:
Good catch